src/platforms/esp32/components/avm_builtins/gpio_driver.c

```diff
@@ -422,7 +422,7 @@ static term gpiodriver_set_int(Context *ctx, int32_t target_pid, term cmd)
 
     struct GPIOData *gpio_data = ctx->platform_data;
 
-    term gpio_num_term = term_to_int32(term_get_tuple_element(cmd, 1));
+    term gpio_num_term = term_get_tuple_element(cmd, 1);
     gpio_num_t gpio_num;
     if (LIKELY(term_is_integer(gpio_num_term))) {
         avm_int_t pin_int = term_to_int32(gpio_num_term);
```

Comment:
Good catch above. I believe this should also be changed to `uint32_t` from `avm_int_t`.

Reply:
setting `pin_int` to `uint32_t` generates following warnings:
```
In file included from /work/AtomVM/src/libAtomVM/memory.h:34,
                 from /work/AtomVM/src/libAtomVM/term.h:37,
                 from /work/AtomVM/src/libAtomVM/ets.h:29,
                 from /work/AtomVM/src/libAtomVM/globalcontext.h:36,
                 from /work/AtomVM/src/libAtomVM/context.h:31,
                 from /work/AtomVM/src/libAtomVM/bif.h:32,
                 from /work/AtomVM/src/platforms/esp32/components/avm_builtins/gpio_driver.c:33:
/work/AtomVM/src/platforms/esp32/components/avm_builtins/gpio_driver.c: In function 'gpiodriver_set_int':
/work/AtomVM/src/platforms/esp32/components/avm_builtins/gpio_driver.c:429:31: warning: comparison of unsigned expression in '< 0' is always false [-Wtype-limits]
  429 |         if (UNLIKELY((pin_int < 0) || (pin_int >= GPIO_NUM_MAX))) {
      |                               ^
/work/AtomVM/src/libAtomVM/utils.h:168:45: note: in definition of macro 'UNLIKELY'
  168 |     #define UNLIKELY(x) __builtin_expect(!!(x), 0)
      |                                             ^
```
So setting it to `int32_t`.

Please see attached changes.